aquadoggo/src/libp2p/config.rs

```diff
@@ -0,0 +1,163 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+
+use std::path::PathBuf;
+
+use anyhow::Result;
+use libp2p::identity::Keypair;
+use libp2p::swarm::ConnectionLimits;
+use libp2p::Multiaddr;
+use log::info;
+use serde::{Deserialize, Serialize};
+
+use crate::libp2p::identity::Identity;
+
+/// Keypair file name.
+const KEYPAIR_FILE_NAME: &str = "libp2p.pem";
+
+/// QUIC default transport port.
+const QUIC_PORT: u16 = 48648;
+
+/// Libp2p config for the node.
+#[derive(Debug, Clone, Deserialize, Serialize)]
+pub struct Libp2pConfiguration {
+    /// Dial concurrency factor.
+    ///
+    /// Number of addresses concurrently dialed for a single outbound
+    /// connection attempt.
+    pub dial_concurrency_factor: u8,
```

Comment:
First I thought this parameter is about "Maximum outgoing connections." but then I realised it is something else, I'm not sure though what. "An attempt to create ONE outgoing connection by using MULTIPLE addresses?". What is dialing in this context? Is there a different way to describe what this config changes?

Reply:
You are attempting to connect with a peer for whom you hold multiple addresses. This value defines the number of addresses you will dial concurrently in an attempt to make a connection with the peer.